File: ios/AppInfo/AppInfo.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(AppInfo, NSObject)
RCT_EXTERN_METHOD(getAppVersion:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)
RCT_EXTERN_METHOD(getApplicationQueriesSchemes:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)
@end

File: ios/AppInfo/AppInfo.swift
import Foundation
import React

@objc(AppInfo)
class AppInfo: NSObject {
  @objc
  static func requiresMainQueueSetup() -> Bool {
    return true
  }

  @objc
  func getAppVersion(_ resolve: RCTPromiseResolveBlock, rejecter reject: RCTPromiseRejectBlock) {
    if let version = Bundle.main.infoDictionary?["CFBundleShortVersionString"] as? String {
      resolve(version)
    } else {
      let error = NSError(domain: "", code: 200, userInfo: nil)
      reject("no_version", "No version found", error)
    }
  }
}
